OpenSim/Common/DelimFileAdapter.h

```diff
@@ -22,19 +22,65 @@
 #ifndef OPENSIM_DELIM_FILE_ADAPTER_H_
 #define OPENSIM_DELIM_FILE_ADAPTER_H_
 
+#include "SimTKcommon.h"
+
 #include "FileAdapter.h"
 #include "TimeSeriesTable.h"
 
+#include <fstream>
+#include <regex>
+
 namespace OpenSim {
 
+class IncorrectNumTokens : public Exception {
+public:
+    IncorrectNumTokens(const std::string& file,
+                       size_t line,
+                       const std::string& func,
+                       const std::string& msg) :
+        Exception(file, line, func) {
+
+        addMessage(msg);
+    }
+};
+
+class DataTypeMismatch : public Exception {
+public:
+    DataTypeMismatch(const std::string& file,
+                     size_t line,
+                     const std::string& func,
+                     const std::string& expected,
+                     const std::string& received) :
+        Exception(file, line, func) {
+        std::string msg = "expected = " + expected;
+        msg += " received = " + received;
+
+        addMessage(msg);
+    }
+};
+
 /** DelimFileAdapter is a FileAdapter that reads and writes text files with
 given delimiters. CSVFileAdapter and MOTFileAdapter derive from this class and
 set the delimiters appropriately for the files they parse. The read/write
 functions return/accept a specific type of DataTable referred to as Table in 
 this class.                                                                   
 Header in the file is assumed to end with string "endheader" occupying a full
 line.                                                                         */
+template<typename T>
 class OSIMCOMMON_API DelimFileAdapter : public FileAdapter {
+    static_assert(std::is_same<T, double           >::value ||
+		  std::is_same<T, SimTK::Vec2      >::value ||
```

Comment:
The implementation currently supports only Vec2, Vec3, Vec4, Vec5 & Vec6, The check here is according to that.

I was initially asked to support Vec3, Vec6 and SpatialVec (as title of this PR says) only and so I wrote explicit specializations to accommodate this. Now you asking me to support any Vec<T>. Not sure what you exactly want and how much of it is justified as use cases. If you want support for any Vec<T>, I will have to change the implementation to partial specializations.